cli/cmd/migrate_webhooks.go

```diff
@@ -0,0 +1,464 @@
+				Version v1beta1 is the new default version and it is highly encouraged to use this one. Version v1alpha1 won't be supported from Keptn 0.19.0`,
+	Args: cobra.NoArgs,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return doMigration(migrateWebhooksParams)
+	},
+}
+
+func doMigration(params *migrateWebhooksCmdParams) error {
+	endPoint, apiToken, err := credentialmanager.NewCredentialManager(assumeYes).GetCreds(namespace)
+	if err != nil {
+		return fmt.Errorf(authErrorMsg)
+	}
+
+	api, err := internal.APIProvider(endPoint.String(), apiToken)
+	if err != nil {
+		return internal.OnAPIError(err)
+	}
+
+	if *params.ProjectName != "" { // project for migration selected
+		project, err := checkProjectExists(*params.ProjectName, api)
+		if err != nil {
+			return fmt.Errorf("Failed to migrate webhook configurations for project %s: %s", *params.ProjectName, err.Error())
+		}
+
+		webhooks, err := getProjectWebhooks(project, api)
+		if err != nil {
+			return fmt.Errorf("cannot retrieve webhook-config for project %s: %s", *params.ProjectName, err.Error())
+		}
+		if err := migrateWebhooks(webhooks, params, api); err != nil {
+			return fmt.Errorf("cannot migrate webhook-config for project %s: %s", *params.ProjectName, err.Error())
+		}
+	} else { // project for migration not selected, migrating all projects
+		projects, err := api.ProjectsV1().GetAllProjects()
+		if err != nil {
+			return fmt.Errorf("cannot retrieve projects: %s", err.Error())
+		}
+
+		for _, project := range projects {
+			webhooks, err := getProjectWebhooks(project, api)
+			if err != nil {
+				return fmt.Errorf("cannot retrieve webhook-config for project %s: %s", project.ProjectName, err.Error())
+			}
+			if err := migrateWebhooks(webhooks, params, api); err != nil {
+				return fmt.Errorf("cannot migrate webhook-config for project %s: %s", project.ProjectName, err.Error())
+			}
+		}
+	}
+
+	return nil
+}
+
+func getProjectWebhooks(project *models.Project, api *api.APISet) ([]*webhookResource, error) {
+	var webhooks []*webhookResource
+
+	// getting project resources
+	projectWebhookResource, err := api.ResourcesV1().GetProjectResource(project.ProjectName, webhookURI)
+	if err != nil && errors.Is(err, ErrResourceNotFound) {
+		return nil, fmt.Errorf("cannot retrieve webhook resource on project level for project %s: %s", project.ProjectName, err.Error())
+	}
+	if projectWebhookResource != nil {
+		tmpWebhookResource := webhookResource{
+			Project:         &project.ProjectName,
+			WebhookResource: projectWebhookResource,
+		}
+		webhooks = append(webhooks, &tmpWebhookResource)
+	}
+
+	// getting stage resources
+	for _, stage := range project.Stages {
+		stageWebhookResource, err := api.ResourcesV1().GetStageResource(project.ProjectName, stage.StageName, webhookURI)
+		if err != nil && errors.Is(err, ErrResourceNotFound) {
+			return nil, fmt.Errorf("cannot retrieve webhook resource on stage level for project %s: %s", project.ProjectName, err.Error())
```

Comment:
I think we should check for `err != nil && !errors.Is(err, ErrResourceNotFound)`, right? otherwise the `getProjectWebhooks()` function would return an empty list of webhooks if there is any stage or service without a webhook

Reply:
In this kind of scenario I would suggest to rather continue with the next stage, instead of immediately returning nil

Reply:
yes fixed :)